src/main/java/com/google/sps/youtube/YoutubeInfoScraper.java

```diff
@@ -0,0 +1,74 @@
+
+package com.google.sps.youtube;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.ChannelListResponse;
+import com.google.api.services.youtube.YouTubeRequestInitializer;
+import com.google.api.services.youtube.model.Channel;
+import com.google.api.services.youtube.model.ChannelContentDetails;
+import com.google.api.services.youtube.model.ChannelContentDetails.RelatedPlaylists;
+import java.util.List;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import com.google.sps.youtube.YoutubeResponse;
+import java.util.Properties; 
+import java.io.FileInputStream;
+
+
+public class YoutubeInfoScraper implements YoutubeResponse {
+
+  // TODO: Add seperate file to hold API Key  
+  private String API_KEY = "";
+  private static final String APPLICATION_NAME = "promotube";
+
+  private YouTube youtube;
+
+  private YoutubeInfoScraper(YouTube youtube){
+      this.youtube = youtube;
+  }
+
+  private YoutubeInfoScraper(){
+    youtube = new YouTube.Builder(new NetHttpTransport(),
+        JacksonFactory.getDefaultInstance(),
+        null)
+        .setApplicationName(APPLICATION_NAME)
+        .setYouTubeRequestInitializer(new YouTubeRequestInitializer(API_KEY))
+        .build();
+  }
+
+  /**
+  * @param ytResponse Pass in this object or any other object that implements YoutubeResponse.
+  * @param channelId Id of a channel.  Can be found in channel's urls.
+  * e.g. https://www.youtube.com/channel/<CHANNEL-ID>
+  * @return string of the channel's upload playlist id.  Null if id is invalid or 
+  * no items were found.
+  */
+  public String scrapeChannelUploadPlaylist(YoutubeResponse ytResponse, String channelId)
+    throws IOException{
+        ChannelListResponse response = ytResponse.getYoutubeChannelListResponse(channelId);
+        if(response==null || response.getItems()==null || response.getItems().size()==0){
+            return "null";
```

Comment:
Returning a string containing "null" is something that needs to be parsed by the client in the event that it actually is "null", so you really almost never want to do this.

You probably want to make this method @Nullable and actually return null, or even better don't use nulls and return Optional<String> and in this case you'd return Optional.empty() since nulls are discouraged in java.